### PR TITLE
Add test case for RuleSchemaMetaData(#6731)

### LINKED ISSUE #6712: Add more test case for HintManager
- [ ] Add test case for addDatabaseShardingValue if `databaseShardingOnly`
- [ ] Add test case for addTableShardingValue if `databaseShardingOnly`

### LINKED ISSUE #6731: Add test case for RuleSchemaMetaData

### PR DESCRIPTION
**Pull request by GeniusQ**
Fixes #6731 

Changes proposed in this pull request:
-Add test case for RuleSchemaMetaData

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/hint/HintManagerTest.java [hunk 1]
```diff
@@ -111,6 +111,32 @@ public void assertIsDatabaseShardingOnlyWithoutSet() {
...
+    @Test
+    public void assertAddDatabaseShardingValueOnlyDatabaseSharding() {
+        try (HintManager hintManager = HintManager.getInstance()) {
+            hintManager.setDatabaseShardingValue(1);
+            assertTrue(HintManager.isDatabaseShardingOnly());
+            hintManager.addDatabaseShardingValue("logic_table", 2);
+            assertFalse(HintManager.isDatabaseShardingOnly());
+            assertThat(HintManager.getDatabaseShardingValues("logic_table").size(), is(1));
+            assertTrue(HintManager.getDatabaseShardingValues("logic_table").contains(2));
+            hintManager.close();
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/hint/HintManagerTest.java:123]: Is it necessary to call `close` again?
`Try with resources` is close the resource.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/hint/HintManagerTest.java [hunk 2]
```diff
@@ -111,6 +111,32 @@ public void assertIsDatabaseShardingOnlyWithoutSet() {
...
+    @Test
+    public void assertAddTableShardingValueOnlyDatabaseSharding() {
+        try (HintManager hintManager = HintManager.getInstance()) {
+            hintManager.setDatabaseShardingValue(1);
+            assertTrue(HintManager.isDatabaseShardingOnly());
+            hintManager.addTableShardingValue("logic_table", 2);
+            assertFalse(HintManager.isDatabaseShardingOnly());
+            assertThat(HintManager.getTableShardingValues("logic_table").size(), is(1));
+            assertTrue(HintManager.getTableShardingValues("logic_table").contains(2));
+            hintManager.close();
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/hint/HintManagerTest.java:136]: Is it necessary to call `close` again?
Try with resources is close the resource.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.assertNotNull;
+
+public class RuleSchemaMetaDataTest {
+    @Test
+    public void assertNewInstance() {
+        RuleSchemaMetaData obj = new RuleSchemaMetaData(null, null);
+        assertNotNull("RuleSchemaMetaData is null", obj);
+    }
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataTest.java:29]: The assertion of new instance does not make sense.
Please add test case for `getSchemaMetaData`

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/hint/HintManagerTest.java (hunk 1 above)
Comment (GeniusQ) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/hint/HintManagerTest.java]: ok, i fix it now.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataTest.java (hunk 3 above)
Comment (GeniusQ) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataTest.java:29]: ok

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataTest.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.apache.shardingsphere.sql.parser.binder.metadata.schema.SchemaMetaData;
+import org.apache.shardingsphere.sql.parser.binder.metadata.table.TableMetaData;
+import org.junit.Test;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertNotNull;
+
+public class RuleSchemaMetaDataTest {
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/RuleSchemaMetaDataTest.java:28]: 1. Please add final if the class is not design for extension;
2. Please keep one blank line between class definition and method.

**Review by terrymanu — APPROVED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**